dashboard/app/assets/javascripts/header.js

```diff
@@ -445,16 +445,28 @@ function mergedActivityCssClass(a, b) {
 }
 
 function populateProgress(scriptName) {
-  var scriptProgress = dashboard.clientState.allLevelsProgress()[scriptName] || {};
+  var clientProgress = dashboard.clientState.allLevelsProgress()[scriptName] || {};
+  $.ajax('/api/user_progress/' + scriptName).done(function (data) {
+    if (!data || !data.levels) {
+      return;
+    }
+    var serverProgress = data.levels;
 
-  // Aggregate progress from server and client
-  $.each(scriptProgress, function (level_id, result) {
-    var level_link = $('#level-' + level_id);
-    var status = mergedActivityCssClass(result, level_link.data('result'));
+    // Aggregate progress from server and client
+    var keys = Object.keys(clientProgress).concat(Object.keys(serverProgress));
```

Comment:
Consider the following which avoids an allocation and copy for all of the keys in clientProgress and serverProgress.  I don't feel strongly though, so if you prefer the current approach that's fine.

var uniqueKeys = {};
Object.keys(clientProgress).forEach(function (key) {
      uniqueKeys[key] = true;
 });
Object.keys(serverProgress).forEach(function (key) {
      uniqueKeys[key] = true;
 });